cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,185 @@
+  private static final String PROJECT_ID = "";
+  private static final String CONNECTION_NAME = "";
+  private static final String DB_NAME = "";
+  private static final String DB_USER = "";
+  private static final String DB_PASSWORD = "";
+
+  private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+  private static final Tracer TRACER = Tracing.getTracer();
+
+  static {
+    try {
+      // Set up StackDriver
+      StackdriverTraceExporter.createAndRegister(
+          StackdriverTraceConfiguration.builder()
+              .setProjectId(PROJECT_ID)
+              .build());
+
+    } catch (java.io.IOException e) {
+      LOGGER.log(Level.WARNING, "Warning: Could not set up Stackdriver Trace", e);
+    }
+
+  }
+
+  @Override
+  public void service(HttpRequest request, HttpResponse response)
+      throws IOException {
+
+    Integer duration = Integer.parseInt(request.getFirstQueryParameter("duration").orElse("5000"));
+    Integer interval = Integer.parseInt(request.getFirstQueryParameter("interval").orElse("500"));
+
+    // Trace every request
+    TraceConfig traceConfig = Tracing.getTraceConfig();
+    traceConfig.updateActiveTraceParams(
+        traceConfig.getActiveTraceParams().toBuilder().setSampler(Samplers.alwaysSample()).build());
+
+    // Set up URL parameters
+    String jdbcURL = String.format("jdbc:postgresql:///%s", DB_NAME);
+    Properties connProps = new Properties();
+    connProps.setProperty("user", DB_USER);
+    connProps.setProperty("password", DB_PASSWORD);
+    connProps.setProperty("socketFactory", "com.google.cloud.sql.postgres.SocketFactory");
+    connProps.setProperty("cloudSqlInstance", CONNECTION_NAME);
+
+    LOGGER.log(Level.INFO, "Started logging.");
+
+    LOGGER.log(Level.INFO, "Started logging.");
+    // Create thread pool
+    ScheduledExecutorService execService = Executors.newScheduledThreadPool(2);
+
+    // Start tests for regular connections
+    execService.scheduleAtFixedRate(() -> {
+      connectRegular(jdbcURL, connProps);
+    }, 0, interval, TimeUnit.MILLISECONDS);
+
+    // Start tests for pooled connections
+    DataSource pool = createPool(jdbcURL, connProps);
+    execService.scheduleAtFixedRate(() -> {
+      connectWithPool(pool);
+    }, 0, interval, TimeUnit.MILLISECONDS);
+
+    try {
+      TimeUnit.MILLISECONDS.sleep(duration);
+    } catch (InterruptedException e) {
+      response.setStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+      BufferedWriter writer = response.getWriter();
+      writer.write(ExceptionUtils.getStackTrace(e));
+      return;
+    }
+
+    execService.shutdown();
+
+    BufferedWriter writer = response.getWriter();
+    writer.write(
+        String.format("Logged connection attempts to Stackdriver Trace for %d ms", duration));
+  }
+
+  private static void connectRegular(String jdbcURL, Properties connProps) {
+    try (Scope rootSpan = TRACER.spanBuilder("regular-connection").startScopedSpan()) {
```

Comment:
Should we add a UUID to these so we can tell the different between different function requests?

Reply:
done.